From: Arnaud Le Blanc <arnaud.lb@gmail.com>
Subject: Update tests for the bailout state corruption fix

Test-only hunks of 3584a98bda0 (PHP-8.4 "Avoid state corruption via bailout")
for Zend/tests/gh11189*.phpt and fibers/get-return-after-bailout.phpt: the
fix no longer lets the first allocation failure be recovered, so the
expected output changes.
---
diff --git a/Zend/tests/fibers/get-return-after-bailout.phpt b/Zend/tests/fibers/get-return-after-bailout.phpt
index 04bd464cfab..7d8e146ef8c 100644
--- a/Zend/tests/fibers/get-return-after-bailout.phpt
+++ b/Zend/tests/fibers/get-return-after-bailout.phpt
@@ -1,12 +1,5 @@
 --TEST--
 Fiber::getReturn() after bailout
---SKIPIF--
-<?php
-$zend_mm_enabled = getenv("USE_ZEND_ALLOC");
-if ($zend_mm_enabled === "0") {
-    die("skip Zend MM disabled");
-}
-?>
 --FILE--
 <?php
 
@@ -16,13 +9,13 @@
 });
 
 $fiber = new Fiber(static function (): void {
-    str_repeat('X', PHP_INT_MAX);
+    var_dump(new class { public function __debugInfo() { return ''; } }); // cause a bailout
 });
 $fiber->start();
 
 ?>
 --EXPECTF--
-Fatal error: Allowed memory size of %d bytes exhausted%s(tried to allocate %d bytes) %sget-return-after-bailout.php on line %d
+Fatal error: __debuginfo() must return an array in %s on line %d
 
 Fatal error: Uncaught FiberError: Cannot get fiber return value: The fiber exited with a fatal error in %sget-return-after-bailout.php:%d
 Stack trace:
diff --git a/Zend/tests/gh11189.phpt b/Zend/tests/gh11189.phpt
index adbc3ce487c..32aa533cbea 100644
--- a/Zend/tests/gh11189.phpt
+++ b/Zend/tests/gh11189.phpt
@@ -8,7 +8,7 @@
 if (getenv("USE_ZEND_ALLOC") === "0") die("skip ZMM is disabled");
 ?>
 --INI--
-memory_limit=2M
+memory_limit=10M
 --FILE--
 <?php
 
@@ -28,5 +28,4 @@
 
 ?>
 --EXPECTF--
-Success
-Fatal error: Allowed memory size of %s bytes exhausted%s(tried to allocate %s bytes) in %s on line %d
+Fatal error: Allowed memory size of 10485760 bytes exhausted %S(tried to allocate %d bytes) in %s on line %d
diff --git a/Zend/tests/gh11189_1.phpt b/Zend/tests/gh11189_1.phpt
index 17b9967bc31..0791ca87d62 100644
--- a/Zend/tests/gh11189_1.phpt
+++ b/Zend/tests/gh11189_1.phpt
@@ -28,5 +28,4 @@
 
 ?>
 --EXPECTF--
-Success
-Fatal error: Allowed memory size of %s bytes exhausted%s(tried to allocate %s bytes) in %s on line %d
+Fatal error: Allowed memory size of 2097152 bytes exhausted %S(tried to allocate %d bytes) in %s on line %d
